### PR TITLE
Shrink2

### PR DESCRIPTION
**Pull request by andrei-marinica**
Second wave of optimizations.

My idea of an early exit for the deserializer does more harm than good, so I turned it off. Will probably remove it altogether in the next version, after further investigations.

Other than that, the endpoint argument handlers were simplified and optimized.

### CODE REVIEW

File: elrond-codec/src/top_de.rs [hunk 1]
```diff
@@ -26,60 +26,89 @@ pub trait TopDecode: Sized {
...
             let bytes = input.into_boxed_slice_u8();
             let bytes_vec = boxed_slice_into_vec(bytes);
             let cast_vec: Vec<T> = unsafe { core::mem::transmute(bytes_vec) };
-            Ok(cast_vec)
+            f(Ok(cast_vec))
         } else {
-            let mut slice = input.get_slice_u8();
-            let mut_slice = &mut slice;
+            let bytes = input.into_boxed_slice_u8();
+            // let mut slice = input.get_slice_u8();
```
Comment (sasurobert) [inline elrond-codec/src/top_de.rs:102]: MY delete commented code

File: elrond-wasm-derive/src/arg_regular.rs [hunk 2]
```diff
@@ -5,71 +5,69 @@ use super::util::*;
...
+                elrond_wasm::load_single_arg::<T, BigInt, BigUint, #arg_ty>(self.api.clone(), #arg_index_expr, #arg_name_expr)
             }
         },
     }
 }
 
 pub fn generate_load_dyn_arg(arg: &MethodArg,
-        loader_expr: &proc_macro2::TokenStream,
-        err_handler_expr: &proc_macro2::TokenStream) -> proc_macro2::TokenStream {
+        loader_expr: &proc_macro2::TokenStream) -> proc_macro2::TokenStream {
```
Comment (sasurobert) [inline elrond-wasm-derive/src/arg_regular.rs:28]: Can you rename proc_macro2?

File: elrond-wasm/src/io/signal_error.rs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+/// An error handler that simply panics whenever `signal_error` is called.
+/// Especially useful for unit tests.
+pub struct PanickingSignalError;
+
+impl SignalError for PanickingSignalError {
+    fn signal_error(&self, message: &[u8]) -> ! {
+        panic!("PanickingDynArgErrHandler panicked: {}", core::str::from_utf8(message).unwrap())
+    }
+}
```
Comment (sasurobert) [inline elrond-wasm/src/io/signal_error.rs:94]: Empty line to end of filr

**Review by sasurobert — APPROVED** [review]

File: elrond-wasm-derive/src/arg_regular.rs (hunk 2 above)
Comment (andrei-marinica) [reply elrond-wasm-derive/src/arg_regular.rs:28]: Not mine, it is part of the macro framework :)

Comment (andrei-marinica) [reply elrond-wasm-derive/src/arg_regular.rs:28]: Wish I could :))

**Pull request merged**